src/main/java/players/search/SearchPlayer.java

```diff
@@ -0,0 +1,114 @@
+package players.search;
+
+import core.*;
+import core.actions.AbstractAction;
+import utilities.Pair;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+
+import static utilities.Utils.noise;
+
+public class SearchPlayer extends AbstractPlayer {
```

Comment:
Search is quite abstract, can we call it something more precise? Or at least add some docs to clarify what type of search this does